### PR TITLE
router: Make direct response body size configurable

### LINKED ISSUE #13422: Give the option to increase the maximun size of the HTTP body in responses to be bigger than 4 KB
I know it is for keeping the proxy’s memory footprint from growing too large but it would help to have the option to use a body bigger than 4 KB for the HTTP responses.

### PR DESCRIPTION
**Pull request by dio**
Commit Message: This patch adds `max_body_size_bytes` to set the maximum bytes of the direct response body size.
Risk Level: Low, since the default behavior is preserved.
Testing: Updated to test the newly introduced config.
Docs Changes: Updated.
Release Notes: Added.
Fixes #13422

Signed-off-by: Dhi Aurrahman <dio@tetrate.io>

### CODE REVIEW

File: source/common/router/config_utility.cc [hunk 1]
```diff
@@ -124,6 +124,8 @@ std::string ConfigUtility::parseDirectResponseBody(const envoy::config::route::v
     return EMPTY_STRING;
   }
   const auto& body = route.direct_response().body();
+  const uint32_t maxBodySize =
```
Comment (zuercher) [inline source/common/router/config_utility.cc:127]: nit: `max_body_size`

**Review by zuercher — COMMENTED** [review]
One minor nit. 

More of an API question, but I wonder if this should be done as a runtime flag rather than an API field?

**Comment by dio** [comment]
I have no strong preference, I can change it to get the value from a specified `runtime_key`. cc. @envoyproxy/api-shepherds

File: source/common/router/config_utility.cc (hunk 1 above)
Comment (dio) [reply source/common/router/config_utility.cc]: Ah sorry, for this. Updated.

**Comment by mattklein123** [comment]
> I have no strong preference, I can change it to get the value from a specified runtime_key. cc. @envoyproxy/api-shepherds

I don't have a strong opinion on whether we make this a full config field or a runtime field. Do we think people will use this limit on a per-route basis or just want a global override? @alyssawilk any thoughts?

**Comment by dio** [comment]
@alyssawilk the use-case presented in https://github.com/envoyproxy/envoy/issues/13422 is to make it larger than the current limit.

**Comment by alyssawilk** [comment]
Fair, same goes for OOM avoidance avoidance :-P  Unless someone asks to configure it per-route, I think we can have one value.

**Comment by htuch** [comment]
One value is fine to start with. I'd vote to make it part of the API as it makes it possible to reason about limits consistently, e.g. when looking at a config like https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge#best-practices-edge

**Comment by mattklein123** [comment]
Config sounds good. We could put it at the top level of the route config, or possible on the HCM config and wire it through to the route.

**Comment by petedmarsh** [comment]
@zuercher I would also like this change to be merged, is there anything I can do to help?

**Comment by mattklein123** [comment]
@dio can you open a fresh PR and let's get this merged?

**Comment by dio** [comment]
Yes, sure. Sorry this was slipped under my radar :(.

**Pull request closed without merging**